### DISCUSSION TITLE
Confusing Thermal Hydraulics Results

### DISCUSSION THREAD

**Discussion by TheBEllis** [idaholab/moose#22029] [Q&A Modules: General] [answered]
Hi MOOSE team,
I'm doing some comparisons between the MOOSE thermal hydraulics module and other methods of solving for outlet temperatures in a counter-current heat exchanger. I have done some comparisons to the NTU method (https://en.wikipedia.org/wiki/NTU_method) (https://www.mathworks.com/help/physmod/hydro/ref/entuheattransfer.html) and some CFD calculations I have done in ANSYS FLUENT. Below I have attached a picture of the cross-section of the heat exchanger I have modelled. The inner channel is referred to as `ch1/fch` in my input file and carries an ideal gas with an inlet temp of 300 K. There is a 5cm thick wall between this channel and the outer channel, referred to as `ch2/fch_outer` in my input file which carries another ideal gas with an inlet temperature of 500 K. The ideal gas values are sort of arbitrary but are made sure to be consistent with the values used in fluent. 

<img src="https://i.imgur.com/zam65ro.png?raw=true" width="300">

Below I have plotted the outlet temperature of the inner channel (`ch1/fch`). My MOOSE results vary with time as it is a transient solve, but my fluent and NTU method results are steady state hence the straight line. I am seeing a very large differential in my results which I can't explain and was wondering if anyone could spot anything wrong with my input file. As the fluent solve is done on a 3D mesh, the plotted result is the average outlet temperature of the inner channel. Do you know if the `FlowChannel1Phase` is supposed to approximate the average temperature of the channel as well, or would it just capture the maximum temperature?

<img src="https://i.imgur.com/4pNfmro.png?raw=true" width="600">

I have attached my input file below, sorry as there is a lot of faff in it where I am using lots of postprocessors and functions to calculate my NTU results, but they can be ignored for the most part. Any help is greatly appreciated as I am really lost on this one!

```
T_in = 300         # K
m_dot_in = 10
press = 1e5         # Pa

# core parameters
wall_thickness = ${units 5. cm -> m}
fch_length = 10   # m
fch_n_elems = 100
fch_dia = 0.5
fch_radius_wall = ${fparse fch_dia/2 + wall_thickness}
convective_HTC = 2000

#counter flow params
fch_outer_dia = 1.0



[GlobalParams]
  initial_p = ${press}
  initial_vel = 0
  initial_T = ${T_in}
  gravity_vector = '0 0 0'
  rdg_slope_reconstruction = full
  closures = simple_closures
[]

[HeatStructureMaterials]
  [steel]
    type = SolidMaterialProperties
    rho = 8050
    k = 45
    cp = 466
  []
[]

[Modules/FluidProperties]
  [fluid_cold]
    type = IdealGasFluidProperties
    molar_mass = 2.897e-2
    gamma = 1.4
    k = 0.025
    mu = 3e-5
  []

  [fluid_hot]
    type = IdealGasFluidProperties
    molar_mass = 4e-3
    gamma = 1.67
    k = 0.156
    mu = 3e-5
  []
[]

[Closures]
  [simple_closures]
    type = Closures1PhaseSimple
  []
[]

[Components]

    [ch1]
        [inlet]
            type = InletMassFlowRateTemperature1Phase
            input = 'ch1/fch:in'
            m_dot = ${m_dot_in}
            T = ${T_in}
        []

        #Flow channel setup
        [fch]
            type = FlowChannel1Phase
            position = '0 0 0'
            orientation = '1 0 0'
            length = ${fch_length}
            n_elems = ${fch_n_elems}
            A = ${fparse pi * fch_dia * fch_dia / 4.}
            D_h = ${fparse fch_dia}
            fp = fluid_cold
            f = 1.6
        []

        [outlet]
            type = Outlet1Phase
            input = 'ch1/fch:out'
            p = ${press}
        []
    []


    [ch2]
        [inlet]
            type = InletMassFlowRateTemperature1Phase
            input = 'ch2/fch_outer:out'
            m_dot = -${m_dot_in}
            T = 500
        []

        #Flow channel setup
        [fch_outer]
            type = FlowChannel1Phase
            position = '0 0 0'
            orientation = '1 0 0'
            length = ${fch_length}
            n_elems = ${fch_n_elems}
            A = ${fparse pi * ((fch_outer_dia * fch_outer_dia/4) - fch_radius_wall * fch_radius_wall)}
            D_h = ${fparse fch_outer_dia - (2*fch_radius_wall)}
            f = 1.6
            fp = fluid_hot
            initial_T = 500
        []

        [outlet]
            type = Outlet1Phase
            input = 'ch2/fch_outer:in'
            p = ${press}
        []
    []

    #Heat structure setup

    [hs]
      type = HeatStructureCylindrical
      position = '0 0 0'
      orientation = '1 0 0'
      inner_radius = ${fparse fch_dia / 2.}
      length = ${fch_length}
      n_elems = ${fch_n_elems}
      materials = 'steel'
      initial_T = ${T_in}
      names = 'block'
      widths = ${fparse wall_thickness}
      n_part_elems = 10
    []

    [HeatStructureHeatTransferOuter]
        type = HeatTransferFromHeatStructure1Phase
        flow_channel = 'ch2/fch_outer'
        hs = hs
        hs_side = outer
        Hw = ${fparse convective_HTC}
        P_hf = ${fparse 2 * fch_radius_wall * pi}
    []

    [HeatStructureHeatTransferInner]
        type = HeatTransferFromHeatStructure1Phase
        flow_channel = 'ch1/fch'
        hs = hs
        hs_side = inner
        Hw = ${fparse convective_HTC}
        P_hf = ${fparse fch_dia * pi}
    []
[]

[Functions]

  [t_out_percentage_diff]
    type = ParsedFunction
    value = '100*(moose_temp-ntu_temp)/moose_temp'
    vars = 'ntu_temp moose_temp'
    vals = 'NTU_t_out fch_T_out'
  []

  [NTU_t_out]
    type = ParsedFunction
    value = '300-(q/C_min)'
    vars = 'C_min q'
    vals = 'C_min q'
  []

  [q]
    type = ParsedFunction
    value = 'q_max*Effectivness'
    vars = 'q_max Effectivness'
    vals = 'q_max Effectivness'
  []

  [q_max]
    type = ParsedFunction
    value = '(C_min) * (300-500)'
    vars = 'C_min NTU'
    vals = 'C_min NTU'
  []

  [Effectivness]
    type = ParsedFunction
    value = '(1-exp(-NTU*(1-Cr)))/(1-Cr*exp(-NTU*(1-Cr)))'
    vars = 'NTU Cr'
    vals = 'NTU Cr'
  []

  [NTU]
    type = ParsedFunction
    value = 'HTC/(C_min)'
    vars = 'C_min HTC'
    vals = 'C_min Total_HTC'
  [../]

  [C_min]
    type = ParsedFunction
    value = 'cp*${m_dot_in}'
    vars = 'cp'
    vals = 'cp_cold'
  []

  [Cr]
    type = ParsedFunction
    value = '(cp_cold*${m_dot_in})/(cp_hot*${m_dot_in})'
    vars = 'cp_cold cp_hot'
    vals = 'cp_cold cp_hot'
  []

  #Heat transfer area of the inner channel
  [HT_Area1]
    type = ParsedFunction
    value = '${fch_length} * pi * ${fch_dia}'
  []

  #Heat transfer area of the outer channel
  [HT_Area2]
    type = ParsedFunction
    value = '${fch_length}*(pi*2*${fch_radius_wall})'
  []

  [Total_HTC]
    type = ParsedFunction
    value = '1/((1/(A1*${convective_HTC})) + (1/(A2*${convective_HTC})) + (${wall_thickness}/45))'
    vars = 'A1 A2'
    vals = 'HT_Area1 HT_Area2'
  []

[]


[Postprocessors]
  [fch_T_out]
    type = SideAverageValue
    boundary = 'ch1/fch:out'
    variable = T
  []

  [cp_cold]
    type = ADElementAverageMaterialProperty
    block = 'ch1/fch'
    mat_prop = 'cp'
  []

  [cp_hot]
    type = ADElementAverageMaterialProperty
    block = 'ch2/fch_outer'
    mat_prop = 'cp'
  []

  [hot_T_out]
    type = SideAverageValue
    boundary = 'ch2/fch_outer:in'
    variable = T
  []

  [NTU_Temp]
    type = FunctionValuePostprocessor
    function = NTU_t_out
    enable = true
  []

  [percentage_diff_vs_moose]
    type = FunctionValuePostprocessor
    function = t_out_percentage_diff
    enable = true
  []

  [dynamic_hyd_outer]
    type = FunctionValuePostprocessor
    function = ${fparse fch_outer_dia - (2*fch_radius_wall)}
    enable = false
  []
[]

[Executioner]
  type = Transient
  start_time = 0
  end_time = 2000
  dt = 10
  steady_state_detection = false
  line_search = basic
  solve_type = NEWTON

  nl_rel_tol = 1e-8
  nl_abs_tol = 1e-1
  nl_max_its = 100

  l_tol = 1e-7
  l_max_its = 1000
  
[]

[Outputs]
    exodus = true
    csv = true
[]
```

**Comment by lindsayad** [comment]
@joshuahansel @licharlot maybe you have some ideas?

**Comment by joshuahansel** [comment, marked as answer]
> The ideal gas values are sort of arbitrary but are made sure to be consistent with the values used in fluent.

I'm not sure about the "arbitrary" part. I believe `gamma` and `molar_mass` are very relevant for this problem, since they determine the relation between internal energy and temperature. It sounds like you've made them consistent with your other runs, but I just want to make sure.

> Do you know if the FlowChannel1Phase is supposed to approximate the average temperature of the channel as well, or would it just capture the maximum temperature?

It approximates the average. More accurately, it approximates the average of the total energy density (and density and momentum density) and then computes temperature at those values.

This may be difficult: I don't see anything obviously wrong with your input, and your results are at least following physical trends. We just don't know why it's skewed toward the other end of the temperature range. This is mostly just a question of heat transfer, rather than flow. The key things I can think to look at are:
1. heat transfer conditions of THM vs. your other codes (make sure they are all doing the same thing: `htc*(T - T_wall)`)
1. EOS of the working fluid (`gamma` and `molar_mass`)
1. cross-sectional area of each channel
1. heated perimeter of each channel `P_hf`

Number 1 is the most important. I already mentioned 2. 3 and 4 look correct.

**Reply by GiudGiud** [reply]
It may be helpful to output more than just one temperature. htc as a function of space for example could be instructive.
cp is a good idea too but since T is different it's likely to differ a little bit

The EOS are indeed very important. Are you using ideal gas across the board here?
